runtime/lua/vim/uv/_meta/uv.lua

```diff
@@ -0,0 +1,2794 @@
+--- with `uv.udp_bind()` it will be bound to `0.0.0.0` (the "all interfaces" IPv4
+--- address) and a random port number.
+---
+---@param  udp                  uv.uv_udp_t
+---@param  data                 uv.buffer
+---@param  host                 string
+---@param  port                 integer
+---@param  callback             uv.udp_send.callback
+---@return uv.uv_udp_send_t|nil success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_send(udp, data, host, port, callback) end
+
+--- Set broadcast on or off.
+---
+---@param  udp        uv.uv_udp_t
+---@param  on         boolean
+---@return 0|nil      success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_set_broadcast(udp, on) end
+
+--- Set membership for a multicast address. `multicast_addr` is multicast address to
+--- set membership for. `interface_addr` is interface address. `membership` can be
+--- the string `"leave"` or `"join"`.
+---
+---@param  udp            uv.uv_udp_t
+---@param  multicast_addr string
+---@param  interface_addr string
+---@param  membership     "leave"|"join"
+---@return 0|nil          success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_set_membership(udp, multicast_addr, interface_addr, membership) end
+
+--- Set the multicast interface to send or receive data on.
+---
+---@param  udp            uv.uv_udp_t
+---@param  interface_addr string
+---@return 0|nil          success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_set_multicast_interface(udp, interface_addr) end
+
+--- Set IP multicast loop flag. Makes multicast packets loop back to local
+--- sockets.
+---
+---@param  udp        uv.uv_udp_t
+---@param  on         boolean
+---@return 0|nil      success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_set_multicast_loop(udp, on) end
+
+--- Set the multicast ttl.
+---
+--- `ttl` is an integer 1 through 255.
+---
+---@param  udp        uv.uv_udp_t
+---@param  ttl        integer
+---@return 0|nil      success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_set_multicast_ttl(udp, ttl) end
+
+--- Set membership for a source-specific multicast group. `multicast_addr` is multicast
+--- address to set membership for. `interface_addr` is interface address. `source_addr`
+--- is source address. `membership` can be the string `"leave"` or `"join"`.
+---
+---@param  udp            uv.uv_udp_t
+---@param  multicast_addr string
+---@param  interface_addr string|nil
+---@param  source_addr    string
+---@param  membership     "leave"|"join"
+---@return 0|nil          success
+---@return uv.error.message|nil err
+---@return uv.error.name|nil err_name
+function uv.udp_set_source_membership(
+  udp,
+  multicast_addr,
+  interface_addr,
+  source_addr,
+  membership
+)
```

Comment:
Stylua insists on making this multiline, but this seems to confuse luals' "meta detection".

(Another argument for increasing line lengths...)